File: SDK-v5/android/gdpr/app/src/main/java/com/localytics/gdpr/SharedPreferencesUtil.java
package com.localytics.gdpr;

import android.content.Context;
import android.content.SharedPreferences;

class SharedPreferencesUtil {

    private static final String PREFS_NAME = "demo-prefs";
    private static final String LOGGED_IN_KEY = "logged-in";

    static boolean isUserLoggedIn(Context c) {
        SharedPreferences prefs = c.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
        return prefs.getBoolean(LOGGED_IN_KEY, false);
    }

    static void setUserLoggedIn(Context c, boolean loggedIn) {
        SharedPreferences prefs = c.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = prefs.edit();
        editor.putBoolean(LOGGED_IN_KEY, loggedIn);
        editor.apply();
    }

}

File: SDK-v5/kotlin/gdpr-places/app/src/main/java/com/localytics/gdpr/SharedPreferencesUtil.kt
package com.localytics.gdpr

import android.content.Context
import android.content.SharedPreferences

internal object SharedPreferencesUtil {

    private val PREFS_NAME = "demo-prefs"
    private val LOGGED_IN_KEY = "logged-in"

    fun isUserLoggedIn(c: Context): Boolean {
        val prefs = c.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        return prefs.getBoolean(LOGGED_IN_KEY, false)
    }

    fun setUserLoggedIn(c: Context, loggedIn: Boolean) {
        val prefs = c.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        val editor = prefs.edit()
        editor.putBoolean(LOGGED_IN_KEY, loggedIn)
        editor.apply()
    }

}
